packages/core/src/connectors/walletConnect.ts

```diff
@@ -92,21 +188,31 @@ export class WalletConnectConnector extends Connector<
     provider.removeListener('chainChanged', this.onChainChanged)
     provider.removeListener('disconnect', this.onDisconnect)
 
-    typeof localStorage !== 'undefined' &&
+    if (this.version === '1' && typeof localStorage !== 'undefined')
+      // Remove local storage session so user can connect again
       localStorage.removeItem('walletconnect')
+    else {
+      provider.removeListener('session_delete', this.onDisconnect)
+      provider.removeListener('display_uri', this.onDisplayUri)
+    }
   }
 
   async getAccount() {
     const provider = await this.getProvider()
-    const accounts = provider.accounts
+    let accounts
+    if (this.version === '1')
+      accounts = (provider as WalletConnectProvider).accounts
+    else accounts = await provider.enable() // TODO: Should we use `.request({ method: 'eth_accounts' })` instead?
```

Comment:
This returns an empty array when calling `isAuthorized` so auto connecting is not working.

Reply:
Updated the `universal-provider` to `2.1.4` where this issue is resolved. Now returns correct the address from its persistence storage after reload